analysis/subgroups_rates_visualise.R

```diff
@@ -0,0 +1,115 @@
+
+# Load libraries & functions ---
+library(here)
+library(readr)
+library(purrr)
+library(dplyr)
+library(tidyr)
+library(lubridate)
+library(jsonlite)
+library(ggplot2)
+## Load json file listing demographics and comorbidities
+config <- fromJSON(here("analysis", "config.json"))
+
+# Import rates ---
+## For each demographic / comorbidity, we need two graphs: one for females;
+## one for males. Each graph has several lines equal to the number of levels of
+## the demographic / comorbidity variable.
+## For sex, we need one graph, with two lines, this is different from the other
+## demographic / comorbidity variables and therefore imported + plotted 
+## separately in this script.
+## Import mortality rates for sex:
+sex_rates_std <- read_csv(file = here("output", 
+                                      "rates",
+                                      "sex_monthly_std.csv"),
+                          col_types = cols("D", "f", "d"))
+# Import the rest of the mortality rates:
+subgroups_vctr <- c(config$demographics, config$comorbidities)
+subgroups_rates_std <- 
+  map(.x = subgroups_vctr,
+      .f = ~ read_csv(file = here("output", 
+                                  "rates",
+                                  paste0(.x,"_monthly_std.csv")),
+                      col_types = cols("D", "f", "f", "d")))
+
+# Plot rates ---
+## make sequence of dates for the y-axis
+dates <- 
+  c("01-03-2020",
+    "01-06-2020",
+    "01-09-2020",
+    "01-12-2020",
+    "01-03-2021",
+    "01-06-2021",
+    "01-09-2021",
+    "01-12-2021") %>%
+  as_date(., format = "%d-%m-%Y")
+## Plot rates for sex:
+sex_plot <- 
+  sex_rates_std %>%
+    ggplot(., aes(date, value_std, group = sex, col = sex)) + 
+      geom_point() +
+      geom_line() + 
+      theme_minimal() +
+      theme(panel.grid.minor.x = element_blank()) +
+      scale_x_date(name = "Calendar Month",
+                   breaks = dates,
+                   date_labels = "%b-%y") +
+      scale_y_continuous(name = "Standardised Risk per 100,000 Individuals") +
+      scale_colour_discrete(name  ="Sex",
+                            labels = c("Female", "Male"))
+## The remaining variables
+## Make a grid with every demographic/comorb variable combined with "F" and
+## "M", needed since for every demographic/comorb variable we need a plot for 
+## sex equal to "F" and "M".
+subgroups_plots_grid <- 
+  expand_grid(subgroups = subgroups_vctr,
+              sex = c("F", "M"))
+## Make plots and save in list
+subgroups_plots <- 
+    map2(.x = subgroups_plots_grid$subgroups,
+         .y = subgroups_plots_grid$sex,
+         .f = ~ subgroups_rates_std[[which(subgroups_vctr == .x)]] %>% filter(sex == .y) %>%
+                  ggplot(., aes(date, value_std, group = get(.x), col = get(.x))) + 
+                  geom_point() + 
+                  geom_line() +
+                  theme_minimal() +
+                  theme(panel.grid.minor.x = element_blank()) +
```

Comment:
looks like you're trying to get a grid panel, when I run this locally I dont get one

Reply:
Hmmm.. grid lines are present when I run the code. `theme_minimal()` should add gridlines (both vertical and horizontal), and I'm deleting the 'minor' grid lines in the line underneath. No idea why you don't seem to have them?

Reply:
I can see hte gridlines now :)

Reply:
Cool!